Add support paths with numbers

fcsonline opened the pull request:
We have discovered that some specific paths are not properly build.

If your JSON content is something like:

```json
{
  "foo": {
       "bar2": "baz"
   }
}
```

The exported json path is `foo.["bar2"]`. This commit solves this issue.

fcsonline committed:
Add support paths with numbers

We have discovered that some specific paths are not properly build.

If your JSON content is something like:

```json
{
  "foo": {
       "bar2": "baz"
   }
}
```

The exported json path is `foo.["bar2"]`. This commit solves this issue.

MaloPolese commented on src/json.path.ts:
```diff
@@ -87,9 +87,14 @@ export function getPropertyPathWithQuotes(
 export function propertyRequiresQuotes(propertyName: jsonc.Segment): boolean {
+  // If the property start with a numbrer we are forced to require quotes
+  if (propertyName.toString().match(/^\d/)) {
+    return true
```
```suggestion
    return true;
```
Just to be consistent with the code base :sweat_smile:

MaloPolese reviewed (changes requested):
Hello, 
Thanks for your PR! :heart: 
Your code looks good to me. However, tests fail with this new rule.

For example, the test: “getJsonPath test characters without escaping from allowed ranges' ” fails.
And the charactersWithoutEscaping variable needs to be updated with numbers now.

Can you adapt the tests to your PR?

fcsonline commented:
Let me fix them

fcsonline commented:
Btw, I had to update the typescript dependency to avoid this error:

```
node_modules/@types/har-format/index.d.ts:258:6 - error TS1023: An index signature parameter type must be either 'string' or 'number'.
```

fcsonline committed:
Fix issue with Typescript compatibility

We had to bump the typescript version to avoid this error:

```
node_modules/@types/har-format/index.d.ts:258:6 - error TS1023: An index signature parameter type must be either 'string' or 'number'.
```

fcsonline commented:
@MaloPolese Sorry, I'm unable to understand those tests 😅 They are testing one single character at a time. My case is different. One single digit, should be escaped, but a string like "foo2" can be plain.

MaloPolese commented:
Hmmm yes I see, the tests are a bit weird. I'll take a look this weekend :+1:

fcsonline commented:
Thank you for handling those tests! 

And thank you again for all your time invested in this extension! 👏
